### PR TITLE
refactor: use primordials in extensions/fetch

### PR DESCRIPTION
**Pull request by satyarohith**
- [x] extensions/fetch/01_fetch_util.js
- [x] extensions/fetch/20_headers.js
- [x] extensions/fetch/21_formdata.js
- [x] extensions/fetch/22_body.js
- [x] extensions/fetch/22_http_client.js (refactor not required)
- [x] extensions/fetch/23_request.js
- [x] extensions/fetch/23_response.js
- [x] extensions/fetch/26_fetch.js

Best reviewable commit by commit.

Towards https://github.com/denoland/deno/issues/11224

### CODE REVIEW

File: extensions/fetch/20_headers.js [hunk 1]
```diff
@@ -52,7 +74,7 @@
    * @param {HeadersInit} object
    */
   function fillHeaders(headers, object) {
-    if (Array.isArray(object)) {
+    if (ArrayIsArray(object)) {
       for (const header of object) {
         if (header.length !== 2) {
           throw new TypeError(
```
Comment (littledivy) [inline extensions/fetch/20_headers.js:80]: Line 80 - `TypeError`

File: extensions/fetch/20_headers.js [hunk 2]
```diff
@@ -79,11 +101,13 @@
     value = normalizeHeaderValue(value);
 
     // 2.
-    if (!HTTP_TOKEN_CODE_POINT_RE.test(name)) {
+    if (!RegExpPrototypeTest(HTTP_TOKEN_CODE_POINT_RE, name)) {
       throw new TypeError("Header name is not valid.");
```
Comment (littledivy) [inline extensions/fetch/20_headers.js:105]: ditto

File: extensions/fetch/20_headers.js [hunk 3]
```diff
@@ -79,11 +101,13 @@
...
     }
     if (
-      value.includes("\x00") || value.includes("\x0A") || value.includes("\x0D")
+      StringPrototypeIncludes(value, "\x00") ||
+      StringPrototypeIncludes(value, "\x0A") ||
+      StringPrototypeIncludes(value, "\x0D")
     ) {
       throw new TypeError("Header value is not valid.");
```
Comment (littledivy) [inline extensions/fetch/20_headers.js:112]: ditto

File: extensions/fetch/20_headers.js [hunk 4]
```diff
@@ -256,7 +284,7 @@
         context: "Argument 1",
       });
 
-      if (!HTTP_TOKEN_CODE_POINT_RE.test(name)) {
+      if (!RegExpPrototypeTest(HTTP_TOKEN_CODE_POINT_RE, name)) {
         throw new TypeError("Header name is not valid.");
```
Comment (littledivy) [inline extensions/fetch/20_headers.js:288]: ditto

File: extensions/fetch/20_headers.js [hunk 5]
```diff
@@ -284,7 +312,7 @@
...
```
Comment (littledivy) [inline extensions/fetch/20_headers.js:316]: ditto ^_^

File: extensions/fetch/20_headers.js [hunk 6]
```diff
@@ -303,7 +331,7 @@
...
```
Comment (littledivy) [inline extensions/fetch/20_headers.js:335]: ditto

File: extensions/fetch/20_headers.js [hunk 7]
```diff
@@ -337,12 +365,13 @@
       value = normalizeHeaderValue(value);
 
       // 2.
-      if (!HTTP_TOKEN_CODE_POINT_RE.test(name)) {
+      if (!RegExpPrototypeTest(HTTP_TOKEN_CODE_POINT_RE, name)) {
         throw new TypeError("Header name is not valid.");
```
Comment (littledivy) [inline extensions/fetch/20_headers.js:369]: ditto

File: extensions/fetch/22_body.js [hunk 8]
```diff
@@ -21,6 +21,14 @@
...
   const { isReadableStreamDisturbed, errorReadableStream, createProxy } =
     globalThis.__bootstrap.streams;
+  const {
+    ArrayBufferIsView,
+    ArrayPrototypePush,
+    JSONParse,
+    ObjectDefineProperties,
+    PromiseResolve,
+    TypedArrayPrototypeSet,
+  } = window.__bootstrap.primordials;
```
Comment (littledivy) [inline extensions/fetch/22_body.js:31]: Add `Uint8Array`

File: extensions/fetch/23_request.js [hunk 9]
```diff
@@ -26,6 +26,15 @@
...
   const { HttpClient } = window.__bootstrap.fetch;
   const abortSignal = window.__bootstrap.abortSignal;
+  const {
+    ArrayPrototypeMap,
+    ArrayPrototypeSlice,
+    ArrayPrototypeSplice,
+    MapPrototypeHas,
+    MapPrototypeGet,
+    MapPrototypeSet,
+    RegExpPrototypeTest,
```
Comment (littledivy) [inline extensions/fetch/23_request.js:36]: Add `TypeError`

File: extensions/fetch/23_response.js [hunk 10]
```diff
@@ -25,6 +25,14 @@
...
     fillHeaders,
   } = window.__bootstrap.headers;
+  const {
+    ArrayPrototypeMap,
+    ArrayPrototypePush,
+    MapPrototypeHas,
+    MapPrototypeGet,
+    MapPrototypeSet,
+    RegExpPrototypeTest,
+  } = window.__bootstrap.primordials;
```
Comment (littledivy) [inline extensions/fetch/23_response.js:35]: Add `TypeError`

**Review by littledivy — CHANGES_REQUESTED** [review]
It's fun reviewing these :-)

**Comment by SimonRask** [comment]
@satyarohith `01_fetch_util.js` needs `TypeError` added :)

Are you not done since it's a draft?

File: extensions/fetch/23_response.js [hunk 11]
```diff
@@ -25,6 +25,15 @@
...
     fillHeaders,
   } = window.__bootstrap.headers;
+  const {
+    ArrayPrototypeMap,
+    ArrayPrototypePush,
+    MapPrototypeHas,
+    MapPrototypeGet,
+    MapPrototypeSet,
+    RegExpPrototypeTest,
+    TypeError,
```
Comment (SimonRask) [inline extensions/fetch/23_response.js:35]: `Symbol`

Comment (SimonRask) [reply extensions/fetch/23_response.js:35]: `RegExp`

Comment (SimonRask) [reply extensions/fetch/23_response.js:35]: `RangeError`

File: extensions/fetch/26_fetch.js [hunk 12]
```diff
@@ -339,7 +345,12 @@
...
       request.body = null;
       for (let i = 0; i < request.headerList.length; i++) {
-        if (REQUEST_BODY_HEADER_NAMES.includes(request.headerList[i][0])) {
+        if (
+          ArrayPrototypeIncludes(
+            REQUEST_BODY_HEADER_NAMES,
+            request.headerList[i][0],
+          )
+        ) {
           request.headerList.splice(i, 1);
```
Comment (SimonRask) [inline extensions/fetch/26_fetch.js:354]: `ArrayPrototypeSplice`

**Review by SimonRask — CHANGES_REQUESTED** [review]
`20_headers.js`:
- line 299 `ArrayPrototypeSplice`
- line 403 `SymbolFor`
- line 411 `SymbolToStringTag`
- line 423 `SymbolIterator`

`21_formdata.js`:
- line 70 `SymbolToStringTag`

`22_body.js`:
- line 300 `ArrayPrototypeMap`
- line 340 `ArrayPrototypeSlice`

`23_request.js`:
- line 387 `SymbolToStringTag`
- line 391 `SymbolFor`

`23_response.js`:
- line 370 `SymbolToStringTag`
- line 374 `SymbolFor`

`26_fetch.js`
- line 210 `PromisePrototypeCatch`
- line 220 `PromisePrototypeCatch`
- line 240 `PromisePrototypeCatch`
- line 309 `ArrayPrototypeFilter`
- line 363 `ArrayPrototypePush`
- line 407 `ArrayPrototypePush`
- line 411 `PromisePrototypeThen`
- line 432 `PromisePrototypeCatch`
- line 475 `StringPrototypeToLowerCase`

**Comment by satyarohith** [comment]
Thanks for the thorough review, @SimonRask! :)

**Review by bartlomieju — APPROVED** [review]
LGTM

**Pull request merged**